APKBUILD.5

```diff
@@ -0,0 +1,488 @@
+.Xr apk 8
+package manager.  It defines metadata such as the name of the package, the
+version information, the source license, and contact information for the
+developer.  It additionally contains the commands needed to build, test, and
+install the package.
+.Pp
+The
+.Nm
+format is similar to a typical shell script; you set pre-defined variables and
+implement pre-defined functions, and the
+.Xr abuild 1
+(or similar) utility will use them to create the package.
+
+
+.Ss Required Variables
+The following variables must be set in all
+.Nm
+files:
+
+.Bl -tag -width Ds
+.It Cm pkgname
+Specifies name of the package.  This is typically the name of the package
+upstream; however, note that all letters must be lowercased.
+.Pp
+Libraries for scripting languages should have a prefix before the library name
+describing the language.  Such prefixes include
+.Em lua- ,
+.Em perl- ,
+.Em py- ,
+and
+.Em rb- .
+Not all languages use prefixes.  For a definitive list, consult the PREFIXES
+file in the root directory of the repository you are using for packaging.
+
+.It Cm pkgver
+Specifies the version of the software being packaged.  The version specified
+must match the pattern
+.Li digit[.digit]*[_suffix[digit]]*
+where
+.Ar digit
+must be a number greater than or equal to zero,
+and
+.Ar suffix
+must be one of
+.Em alpha ,
+.Em beta ,
+.Em pre ,
+.Em rc ,
+.Em cvs ,
+.Em svn ,
+.Em git ,
+.Em hg ,
+or
+.Em p .
+
+.It Cm pkgrel
+Specifies the package release number of this particular package version.  This
+indicates when a package has changed without a corresponding change in version.
+Always increment
+.Cm pkgrel
+when you change the contents or dependencies of a package.  The first release
+of a package is always
+.Li 0 .
+
+.It Cm pkgdesc
+Specifies what the package contains.
+.Cm pkgdesc
+must be 128 characters or less, and should concisely describe what actions the
+software or items being packaged will allow the user to perform.  For example,
+.Li Dq Fully-featured word processor with spell check and many plugins
+would be a sufficient
+.Cm pkgdesc
+for AbiWord.
+
+.It Cm url
+Specifies the Web address of the package's upstream.  This allows users and
+future maintainers to find documentation, release information, and contact
+information for the package.  If no Web address is available for the package,
+you must set
+.Cm url
+to an empty string ("").
+
+.It Cm license
+Specifies the license under which the package is distributed.  The value
+provided must match a SPDX license identifier.
+
+.It Cm source
+Specifies the location of both local and remote source files used to build the
+package.  Typically, the remote source file(s) or archive(s) is specified,
+followed by any local patches, install scripts, configuration files, or other
+necessary files.
+.El
+
+
+.Ss Optional Variables
+The following variables are not required, but may be set in any
+.Nm
+file:
+
+.Bl -tag -width Ds
+.It Cm arch
```

Comment:
IIRC arch is required.

Reply:
Sorry, it appeared from wiki and `abuild` code that if arch wasn't specified it would default to `$CARCH`.  Looks like that is for subpackages, not actual packages.  I will fix this.

Reply:
Subpackages by default inherit `arch` from the origin package. Basically all variables are just optional for subpackages, but not all are implicitly inherit from the origin. Also some variables should not be declared in subpackage at all (e.g. makedepends or install).